sematic/db/models/external_resource.py

```diff
@@ -133,7 +134,16 @@ def from_resource(cls, resource: ExternalResourceDataclass) -> "ExternalResource
         )
 
     def get_resource_type(self) -> Type[ExternalResourceDataclass]:
-        return type_from_json_encodable(self.type_serialization)
+        try:
+            return type_from_json_encodable(self.type_serialization)
+        except Exception:
+            type_name = self.type_serialization["type"][1]
+            import_path = self.type_serialization["type"][2]["import_path"]
+
+            # We are unable to load the type for the external resource.
+            # This means the type must not be present, which is tantamount
+            # to a missing plugin for that external resource type.
+            raise MissingPluginError(f"{import_path}.{type_name}")
```

Comment:
Please only do this on `ImportError`. All other error types should be bubbled-up as-is.

Reply:
will do